pkg/controller/common/test/manager.go

```diff
@@ -251,6 +251,8 @@ func NewUpdateSimulator(tracker clienttesting.ObjectTracker) clienttesting.React
 		// XXX: update fields that would get modified by the api server on an update
 		accessor.SetResourceVersion(fmt.Sprintf("%d", rand.Int()))
 		if accessor.GetDeletionTimestamp() == nil && len(updateAction.GetSubresource()) == 0 {
+			// XXX: this should only be done if the .spec field actually changes.
+			// i don't think we should do this if finalizers, annotations, labels, etc. are updated
```

Comment:
Correct. Why not add a simple `if !reflect.DeepEquals(oldSpec, newSpec)`?

Reply:
It's not quite that simple, as we're only using `runtime.Object`, so we'd need to use reflection to get the `spec` values.  I can look at adding this at some point in the future.

Reply:
I addressed this too.